upb/msg_internal.h

```diff
@@ -42,9 +42,34 @@ typedef struct {
   int16_t presence;       /* If >0, hasbit_index.  If <0, ~oneof_index. */
   uint16_t submsg_index;  /* undefined if descriptortype != MESSAGE or GROUP. */
   uint8_t descriptortype;
-  uint8_t label;          /* google.protobuf.Label or _UPB_LABEL_* above. */
+  int8_t mode;            /* upb_fieldmode, with flags from upb_labelflags */
 } upb_msglayout_field;
 
+typedef enum {
```

Comment:
why is this one typedef enum and the one below is simply enum?

Reply:
`typedef enum` allows the type to be referred to directly, eg. `upb_fieldmode`. C would otherwise require using the `enum` tag, eg. `enum upb_fieldmode`, which is unnecessarily verbose.

The second isn't typedef'd because we never use the `upb_labelflags` type for any variables or in any function signatures. We're just using it as a bag of constants -- really the only reason to use `enum` at all here is to avoid using a preprocessor `#define` while still allowing constant evaluation.